File: src/test/java/sds/offre/concept_metier/PrixTest.java
package sds.offre.concept_metier;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

class PrixTest {

    @Test
    void doit_créer_un_prix_positif() {

        assertThat(
                Prix.de(34)
        ).isEqualTo(
                Prix.de(34));

    }

    @Test
    void doit_pas_créer_un_prix_négatif() {

        assertThatThrownBy(() ->
                Prix.de(-34)
        ).isInstanceOf(
                IllegalArgumentException.class
        );

    }
}

File: src/test/java/sds/offre/concept_metier/PrixTest.kt
package sds.offre.concept_metier

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test

internal class PrixTest {

    @Test
    fun `doit créer un prix positif`() {
        assertThat(Prix(34))
                .isEqualTo(Prix(34))

    }

    @Test
    fun `doit pas créer un prix négatif`() {
        assertThatThrownBy { Prix(-34) }
                .isInstanceOf(IllegalArgumentException::class.java)
    }
}
